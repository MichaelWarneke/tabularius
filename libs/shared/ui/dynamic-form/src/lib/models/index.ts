import {
  FormControl,
  ValidatorFn,
  AbstractControlOptions,
  AsyncValidatorFn,
  FormArray
} from '@angular/forms';

export class FormControlBase extends FormControl {
  controlType = '';
  key = '';
  label = '';
  maxWidth = 100;
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlTextbox extends FormControlBase {
  controlType = 'input';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlDate extends FormControlBase {
  controlType = 'datepicker';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlCheckbox extends FormControlBase {
  controlType = 'checkbox';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlSelect extends FormControlBase {
  controlType = 'select';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlRadioButton extends FormControlBase {
  controlType = 'radio-button';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export class FormControlArray extends FormArray {
  controlType = 'array';
  key = '';
  label = '';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export { Validators } from '@angular/forms';
