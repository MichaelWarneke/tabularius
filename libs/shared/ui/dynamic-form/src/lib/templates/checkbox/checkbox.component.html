<div class="tabu-checkbox" [formGroup]="group">
  <mat-checkbox [formControlName]="control?.key">{{
    control?.label
  }}</mat-checkbox>
</div>
