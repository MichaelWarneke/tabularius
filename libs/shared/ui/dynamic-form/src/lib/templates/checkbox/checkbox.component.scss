.tabu-checkbox {
  margin: 10px;
}
