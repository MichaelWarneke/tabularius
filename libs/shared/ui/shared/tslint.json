{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
