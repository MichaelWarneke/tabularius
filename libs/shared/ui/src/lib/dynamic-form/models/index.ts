export { QuestionBase } from './question-base';
export { DropdownQuestion } from './question-dropdown';
export { TextboxQuestion } from './question-textbox';
export { DatepickerQuestion } from './question-datepicker';
