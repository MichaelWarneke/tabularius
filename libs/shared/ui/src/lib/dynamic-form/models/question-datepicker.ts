import { QuestionBase } from './question-base';

export class DatepickerQuestion extends QuestionBase<string> {
  controlType = 'datepicker';
  type: Date;

  constructor(options: any = {}) {
    super(options);
    this.type = options['type'] || '';
  }
}
