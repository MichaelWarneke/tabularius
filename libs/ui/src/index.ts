export { TabuAppSidenavModule } from './lib/nav';
