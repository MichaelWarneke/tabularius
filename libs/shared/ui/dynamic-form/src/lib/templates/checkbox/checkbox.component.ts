import { Component } from '@angular/core';
import { FormControlCheckbox } from '../../models';
import { FormGroup } from '@angular/forms';

@Component({
  selector: 'tabu-dynamic-form-checkbox',
  templateUrl: './checkbox.component.html',
  styleUrls: ['./checkbox.component.scss']
})
export class CheckboxComponent {
  control: FormControlCheckbox | null = null;
  group: FormGroup | null = null;

  constructor() {}
}
