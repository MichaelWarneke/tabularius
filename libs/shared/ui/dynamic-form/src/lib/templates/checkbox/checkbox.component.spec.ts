import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { CheckboxComponent } from './checkbox.component';
import { SharedModule } from '@tabularius/shared/ui';
import { FormGroup, FormControl } from '@angular/forms';
import { FormControlCheckbox } from '../../models';
import { By } from '@angular/platform-browser';

describe('CheckboxComponent', () => {
  let component: CheckboxComponent;
  let fixture: ComponentFixture<CheckboxComponent>;
  const key = 'testKey';
  const label = 'testLabel';

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [CheckboxComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CheckboxComponent);
    component = fixture.componentInstance;
    component.group = new FormGroup({ testKey: new FormControl('') });
    component.control = new FormControlCheckbox();
    component.control.key = key;
    component.control.label = label;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain the label', () => {
    const text = fixture.debugElement.query(By.css('mat-checkbox'))
      .nativeElement.innerHTML;
    expect(text).toContain(label);
  });

  it('should contain the formControlName in attributes', () => {
    const attributes = fixture.debugElement.query(By.css('mat-checkbox'))
      .attributes;
    const text = attributes['ng-reflect-name'];
    expect(text).toContain(key);
  });
});
