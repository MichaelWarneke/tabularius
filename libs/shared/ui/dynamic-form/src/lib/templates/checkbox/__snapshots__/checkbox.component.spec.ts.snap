// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxComponent should render 1`] = `
<tabu-dynamic-form-checkbox
  control={[Function FormControlCheckbox]}
  group={[Function FormGroup]}
>
  <div
    class="ng-untouched ng-pristine ng-valid"
    ng-reflect-form="[object Object]"
  >
    <mat-checkbox
      class="tabu-checkbox mat-checkbox mat-accent ng-untouched ng-pristine ng-valid"
      id="mat-checkbox-4"
      ng-reflect-name="testKey"
    >
      <label
        class="mat-checkbox-layout"
        for="mat-checkbox-4-input"
      >
        <div
          class="mat-checkbox-inner-container"
        >
          <input
            aria-checked="false"
            class="mat-checkbox-input cdk-visually-hidden"
            id="mat-checkbox-4-input"
            tabindex="0"
            type="checkbox"
          />
          <div
            class="mat-checkbox-ripple mat-ripple"
            matripple=""
            ng-reflect-animation="[object Object]"
            ng-reflect-centered="true"
            ng-reflect-disabled="false"
            ng-reflect-radius="20"
            ng-reflect-trigger="[object HTMLLabelElement]"
          >
            <div
              class="mat-ripple-element mat-checkbox-persistent-ripple"
            />
          </div>
          <div
            class="mat-checkbox-frame"
          />
          <div
            class="mat-checkbox-background"
          >
            <svg
              class="mat-checkbox-checkmark"
              focusable="false"
              version="1.1"
              viewBox="0 0 24 24"
              xml:space="preserve"
            >
              <path
                class="mat-checkbox-checkmark-path"
                d="M4.1,12.7 9,17.6 20.3,6.3"
                fill="none"
                stroke="white"
              />
            </svg>
            <div
              class="mat-checkbox-mixedmark"
            />
          </div>
        </div>
        <span
          class="mat-checkbox-label"
        >
          <span
            style="display:none"
          >
            Â 
          </span>
          testLabel
        </span>
      </label>
    </mat-checkbox>
  </div>
</tabu-dynamic-form-checkbox>
`;
